test/forward.jl

```diff
@@ -90,4 +90,46 @@ end
     end
 end
 
+
+@testset "structs" begin
+    struct IDemo
+        x::Float64
+        y::Float64
+    end
+
+    function foo(a)
+        obj = IDemo(2.0, a)
+        return obj.x * obj.y
+    end
+
+    let var"'" = Diffractor.PrimeDerivativeFwd
+        @test foo'(100.0) == 2.0
+        @test foo''(100.0) == 0.0
```

Comment:
this second derivative is actually broken on main

Reply:
do we know what change broke it?